Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/distance.h

```diff
@@ -116,67 +127,457 @@ sample_triangles(const TriangleRange& triangles, double distance, OutputIterator
+    Distance_computation<Tree, typename Kernel::Point_3> f(tree, hint, sample_points, &distance);
+    tbb::parallel_for(tbb::blocked_range<std::size_t>(0, sample_points.size()), f);
+    return distance;
+  }
+  else
+#endif
   {
-    double d = CGAL::sqrt( tree.squared_distance(pt) );
-    // if ( d > 1e-1 ) std::cout << pt << "\n";
-    if (d>hdist) hdist=d;
+    double hdist = 0;
+    BOOST_FOREACH(const typename Kernel::Point_3& pt, sample_points)
+    {
+      hint = tree.closest_point(pt, hint);
+      typename Kernel::FT dist = squared_distance(hint,pt);
+      double d = to_double(CGAL::approximate_sqrt(dist));
+      if(d>hdist)
+        hdist=d;
+    }
+    return hdist;
   }
+}
+
+template <class Concurrency_tag, class Kernel, class TriangleMesh,
+          class NamedParameters,
+          class VertexPointMap >
+double approximated_Hausdorff_distance(
+   const TriangleMesh& m1,
+   const TriangleMesh& m2,
+   double precision,
+   NamedParameters np,
+   VertexPointMap vpm,
+   Sampling_method method = RANDOM_UNIFORM
+)
+{
+  std::vector<typename Kernel::Point_3> sample_points;
+  sample_triangle_mesh<TriangleMesh, typename Kernel::Point_3, NamedParameters>(m1, precision ,sample_points, np, method );
+  return approximated_Hausdorff_distance<Concurrency_tag, Kernel>(sample_points, m2, vpm);
+}
+
+// documented functions
+/**
+ * \ingroup PMP_distance_grp
+ * computes the approximated Hausdorff distance of `tm1` from `tm2` by
+ * generating a uniform random point sampling on `tm1`, and by then
+ * returning the distance of the furthest point from `tm2`.
+ *
+ * A parallel version is provided and requires the executable to be
+ * linked against the <a href="http://www.threadingbuildingblocks.org">Intel TBB library</a>.
+ * To control the number of threads used, the user may use the `tbb::task_scheduler_init` class.
+ * See the <a href="http://www.threadingbuildingblocks.org/documentation">TBB documentation</a>
+ * for more details.
+ *
+ * @tparam Concurrency_tag enables sequential versus parallel algorithm.
+ *                         Possible values are `Sequential_tag`
+ *                         and `Parallel_tag`.
+ * @tparam TriangleMesh a model of the concept `FaceListGraph`
+ *         for `CGAL::vertex_point_t`
+ * @tparam NamedParameters1 a sequence of \ref namedparameters for `tm1`
+ * @tparam NamedParameters2 a sequence of \ref namedparameters for `tm2`
+ *
+ * @param tm1 the triangle mesh that will be sampled
+ * @param tm2 the triangle mesh to compute the distance to
+ * @param precision the number of points per squared area unit for the random sampling
+ * @param np1 optional sequence of \ref namedparameters for `tm1` among the ones listed below
+ * @param np2 optional sequence of \ref namedparameters for `tm2` among the ones listed below
+ *
+ * \cgalNamedParamsBegin
+ *    \cgalParamBegin{vertex_point_map} the property map with the points associated to the vertices of `tm1` or `tm2` \cgalParamEnd
```

Comment:
Add _If this parameter is omitted, an internal property map for `CGAL::vertex_point_t` should be available in TriangleMesh_ and in all places where `vertex_point_map` is used.